stdlib/public/runtime/HeapObject.cpp

```diff
@@ -228,6 +228,20 @@ OpaqueValue *swift::swift_projectBox(HeapObject *o) {
   return metadata->project(o);
 }
 
+struct _SwiftEmptyBoxStorage {
+  struct HeapObject header;
+};
+
+swift::HeapLocalVariableMetadata _emptyBoxStorageMetadata;
+
+/// The signleton empty box storage object.
+struct _SwiftEmptyBoxStorage _EmptyBoxStorage = {
```

Comment:
Does this need to be exported or not? If so, it should probably be SWIFT_RUNTIME_STDLIB_INTERFACE; if not, it should probably be in an anonymous namespace.

(Also, no need for the `struct` here.)